flocker/common/test/test_interface.py

```diff
@@ -0,0 +1,133 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for ``flocker.common.interface_decorator``.
+"""
+
+from twisted.trial.unittest import SynchronousTestCase
+
+from eliot.testing import (
+    assertHasMessage, capture_logging
+)
+from eliot import Field, MessageType
+
+from zope.interface import Interface, implementer
+
+from ..common import interface_decorator
+
+
+# Eliot structures for testing ``interface_decorator``.
+METHOD = Field.for_types(
+    u"method", [unicode],
+    u"The name of the decorated method.")
+TEST_MESSAGE = MessageType(u"flocker:common:test:interface:message",
+                           [METHOD])
+TEST_EXCEPTION = MessageType(u"flocker:common:test:interface:exception",
+                             [METHOD])
+
+
+class IDummy(Interface):
+    """
+    Dummy interface with two test methods.
+    """
+    def return_method():
+        """
+        Return something.
+        """
+
+    def raise_method():
+        """
+        Raise something.
+        """
+
+
+@implementer(IDummy)
+class Dummy(object):
+    """
+    Dummy class that implements ``IDummy`` interface.
+    Implements two methods: one to return an object,
+    and the second to raise an ``Exception``.
```

Comment:
Looks like your right margin might be too narrow. 
We all try and wrap at 79 columns ....and it reminds me that I need to fix this in my own editor.